Use local image's application version for upgrade

a-palchikov opened the pull request:
## Description
<!--Required. Provide high-level overview of what the change is for.-->
* Default to the version of the application from local cluster image for upgrades.
* Update latest planet to include a version fix.
* Update versions of (non-)LTS base versions used in robotest.

## Type of change
<!--Required. Keep only those that apply.-->

* Bug fix (non-breaking change which fixes an issue)
* This change has a user-facing impact

## Linked tickets and other PRs
* Ports https://github.com/gravitational/gravity/pull/1642
* Requires https://github.com/gravitational/planet/pull/659.

## TODOs
<!--Required. Keep only those that apply and check them off as they get completed.-->

- [x] Self-review the change
- [x] Perform manual testing
- [ ] Address review feedback

## Implementation
<!--Optional. Add any relevant implementation details that might help the reviewers.-->

## Testing done
<!--Required. Explain what kind of testing these changes underwent.-->


## Additional information
<!--Optional. Anything else that may be relevant.-->

a-palchikov committed:
Default to the version of the application from local cluster image for upgrades.
Also remove 6.1.x from the list of directly upgradable versions due to
Kubernetes version skew.

a-palchikov committed:
Bump planet to latest release

wadells commented on build.assets/robotest_run_suite.sh:
```diff
@@ -7,10 +7,13 @@ readonly UPGRADE_FROM_DIR=${1:-$(pwd)/../upgrade_from}
-UPGRADE_MAP[6.3.6]="ubuntu:18"
+UPGRADE_MAP[6.3.18]="ubuntu:18"
+
+# latest patch release on supported LTS version, keep this up to date
+UPGRADE_MAP[7.0.6]="ubuntu:18"
```
Is 7.0 LTS?  It seems like this comment should be left out and 7.0.6 should fall under "supported non-LTS version" along with 6.3.

a-palchikov committed:
Address review comments

r0mant reviewed (commented):
How is this going to work if `gravity upgrade` is launched outside the installer tarball (which is a valid use-case if, say, the new version was downloaded to the cluster using automatic updates or uploaded earlier)?

a-palchikov committed:
Revise the search for upgrade package as following:
 - use the package pattern if specified
 - attempt to read package version from the tarball directory
 - use the latest version of the installed application (must be
   available in cluster package store)

a-palchikov commented:
The upcoming intermediate upgrades port will supersede this.
